servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/SslContextFactory.java

```diff
@@ -168,21 +195,40 @@ private static void configureTrustManager(SslConfig config, SslContextBuilder bu
         }
     }
 
+    /**
+     * Configures TLS Certificate Compression if enabled and available.
+     * <p>
+     * Note that in addition to the application actually enabling and configuring TLS certificate compression on the
+     * {@link SslConfig}, it must also be available in the environment. Right now it is only supported through
+     * BoringSSL and as such the code checks if the {@link OpenSslContextOption} is available at runtime. If it is not,
+     * no error is raised but the feature is not enabled. This is by design, since certificate compression is a pure
+     * optimization instead of a security feature.
+     *
+     * @param config the ServiceTalk TLS config which enables and configures cert compression.
+     * @param builder netty's builder for the SSL context.
+     * @param nettySslProvider the (potentially null) SSL provider used with netty.
+     * @param forServer if this is for a server or client context.
+     */
     private static void configureCertificateCompression(SslConfig config, SslContextBuilder builder,
                                                         @Nullable io.netty.handler.ssl.SslProvider nettySslProvider,
                                                         boolean forServer) {
         final List<CertificateCompressionAlgorithm> algorithms = config.certificateCompressionAlgorithms();
-        if (algorithms == null || algorithms.isEmpty()) {
+        if (algorithms == null || algorithms.isEmpty() || SSL_PROVIDER_OPTION_SUPPORTED == null) {
             return;
         }
 
         if (nettySslProvider == null) {
             nettySslProvider = forServer ? SslContext.defaultServerProvider() : SslContext.defaultClientProvider();
         }
 
-        // TODO: change once https://github.com/netty/netty/pull/13145 is merged and released
-        if (nettySslProvider != SslProvider.OPENSSL) {
-            return;
+        try {
+            boolean compressionSupported = (boolean) SSL_PROVIDER_OPTION_SUPPORTED.invokeExact(
```

Comment:
Consider using this method only when it's available, otherwise fallback to the pre-existing check.